microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/jaxrs/java/JaxRsCodeLensParticipant.java

```diff
@@ -156,4 +114,69 @@ private static boolean isServerAvailable(String host, int port, int timeout) {
 			return false;
 		}
 	}
+
+	/**
+	 * Returns the provider that can provide JAX-RS method info for the given class,
+	 * or null if no provider can provide info.
+	 *
+	 * @param typeRoot the class to collect JAX-RS method info for
+	 * @param monitor the progress monitor
+	 * @return the provider that can provide JAX-RS method info for the given class,
+	 *         or null if no provider can provide info
+	 */
+	private static IJaxRsInfoProvider getProviderForType(ITypeRoot typeRoot, IProgressMonitor monitor) {
+		for (IJaxRsInfoProvider provider : JavaFeaturesRegistry.getInstance().getJaxRsInfoDefinitions()) {
+			if (provider.canProvideJaxRsMethodInfoForClass(typeRoot, monitor)) {
+				return provider;
+			}
+		}
+		IJaxRsInfoProvider defaultProvider = DefaultJaxRsInfoProvider.getInstance();

```

Comment:
To avoid having specific code with DefaultJaxRsInfoProvider here, I will add new DefaultJaxRsInfoProvider() at the end of the list of getJaxRsInfoDefinitions once all IJaxRsInfoProviders are loaded by extension point.